py/desispec/io/util.py

```diff
@@ -213,16 +213,23 @@ def write_bintable(filename, data, header=None, comments=None, units=None,
 
     hdu = None
     #- Convert data as needed
-    if isinstance(data, (np.recarray, np.ndarray, Table)):
+    if isinstance(data, Table):
+        log.debug("data is astropy.table.Table.")
+        outdata = data
+    elif isinstance(data, (np.recarray, np.ndarray)):
+        log.debug("data is numpy array.")
         outdata = Table(data)
```

Comment:
Admittedly this code pre-dates this PR but for consideration, let's use `outdata = Table(data, copy=False)`. We don't modify the input data as part of this function so we don't need to make a second in-memory copy here, especially given that `astropy.io.fits.convenience.table_to_hdu` below will make yet-another copy.